infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -1008,11 +1046,24 @@ public LogHeader getLogHeader() {
         private final FileChannel pendingTrimChannel;
         @NonNull
         private String fileName;
+
         private Map<Long, AddressMetaData> knownAddresses = new ConcurrentHashMap();
         private Set<Long> trimmedAddresses = Collections.newSetFromMap(new ConcurrentHashMap<>());
         private Set<Long> pendingTrims = Collections.newSetFromMap(new ConcurrentHashMap<>());
+        private volatile int refCount = 0;
 
 
+        public synchronized void retain() {
+            refCount++;
+        }
+
+        public synchronized void release() {
```

Comment:
You are right, I do that intentionally. The rational is, by the time the segment is closed, the ref count should be 0 and if it is not, then "forcefully" close the segment. Close will take precedence over waiting for the refcount to reach 0.